addon-test-support/@ember/test-helpers/validate-error-handler.js

```diff
@@ -5,7 +5,33 @@ const INVALID = Object.freeze({
   message: 'error handler should have re-thrown the provided error',
 });
 
-export default function(callback = Ember.onerror) {
+/**
+  Validate the provided error handler to confirm that it properly re-throws
+  errors when `Ember.testing` is true.
+
+  This is intended to be used by test framework hosts (or other libraries) to
+  ensure that `Ember.onerror` is properly configured. Without a check like
+  this, `Ember.onerror` could _easily_ swallow all errors and make it _seem_
+  like everything is just fine (and have green tests) when in reality
+  everything is on fire...
+
+  @example <caption>Example implementation for `ember-qunit`</caption>
+
+  ```js
+  import { validateErrorHandler } from '@ember/test-helpers';
+
+  test('Ember.onerror is functioning properly', function(assert) {
+    let result = validateErrorHandler();
+    assert.ok(result.isValid, result.message);
+  });
+  ```
```

Comment:
I would prefer it if the `@example` block would be moved below the function signature documentation

Reply:
Totally agreed. Will update.